fido/exceptions.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+
+
+class BaseTimeoutError(Exception):
```

Comment:
I'd just call this one TimeoutError, it's also what other packages typically call their base timeout exception.